### PR TITLE
Dropdown component

### PR DESCRIPTION
**Pull request by ruslanpascoal2**
https://github.com/pyroscope-io/pyroscope/issues/548

Created a dropdown component and a storybook story for it.

### CODE REVIEW

File: stories/Dropdown.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React from 'react';
+import { ComponentStory, ComponentMeta } from '@storybook/react';
+import Dropdown from '@ui/Dropdown';
+import Button from '@ui/Button';
+import { MenuButton, MenuItem, SubMenu } from '@szhsin/react-menu';
+
+const Template: ComponentStory<any> = (args) => (
```
Comment (eh-am) [inline stories/Dropdown.stories.tsx:8]:
```suggestion
const Template: ComponentStory<typeof Dropdown> = (args) => (
```

File: stories/Dropdown.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  <Dropdown buttonText="Dropdown" theming="dark" {...args}>
+    <MenuItem>Option</MenuItem>
+  </Dropdown>
+);
+
+export default {
+  title: 'Components/Dropdown',
+  component: Dropdown,
+} as ComponentMeta<any>;
```
Comment (eh-am) [inline stories/Dropdown.stories.tsx:17]:
```suggestion
} as ComponentMeta<typeof Dropdown>;
```

File: stories/Dropdown.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      <MenuItem>Item 1.1</MenuItem>
+      <MenuItem>Item 1.2</MenuItem>
+    </SubMenu>
+    <SubMenu label="Menu 2">
+      <MenuItem>Item 1.1</MenuItem>
+      <MenuItem>Item 1.2</MenuItem>
+    </SubMenu>
+  </Dropdown>
+);
+
```
Comment (eh-am) [inline stories/Dropdown.stories.tsx:36]: Can we have an example where it behaves like a dropdown, where after you select a value, it replaces the "Menu" there?

**Review by eh-am — CHANGES_REQUESTED** [review]
Beautiful!

---
(Talking to myself)

I am even thinking bastardizing the submenu to make it become like some kind of explanation like grafana does
image

File: stories/Dropdown.stories.tsx (hunk 2 above)
Comment (ruslanpascoal2) [reply stories/Dropdown.stories.tsx:17]: Forgot to replace the any, sorry

**Review by eh-am — APPROVED** [review]

**Comment by eh-am** [comment]
I love it! Really nice touch adding the label as Header, so after I pick a value I still know what's the Dropdown for (Otherwise just having the list of elements don't tell me anything).

Peek 2021-11-19 14-45

**Review by petethepig — APPROVED** [review]
Awesome, thank you for the contribution @ruslanpascoal2 !

**Pull request merged**